modules/videoio/src/cap_mjpeg_decoder.cpp

```diff
@@ -40,13 +40,733 @@
+    MotionJpegCapture(const String&);
+
+    bool open(const String&);
+    void close();
+protected:
+
+    typedef std::deque< std::pair<uint32_t, uint32_t> > frame_list;
+    typedef frame_list::iterator frame_iterator;
+
+    void skipJunk(RiffChunk& chunk, MjpegInputStream& in_str);
+    void skipJunk(RiffList& list, MjpegInputStream& in_str);
+    bool parseRiff(MjpegInputStream& in_str);
+    bool parseAvi(MjpegInputStream& in_str);
+    bool parseHdrlList(MjpegInputStream& in_str);
+    bool parseIndex(MjpegInputStream& in_str, uint32_t index_size);
+    bool parseMovi(MjpegInputStream& in_str);
+    bool parseStrl(MjpegInputStream& in_str, uint8_t stream_id);
+    bool parseInfo(MjpegInputStream& in_str);
+    inline uint32_t getFramePos() const;
+    std::vector<char> readFrame(frame_iterator it);
+
+    MjpegInputStream m_file_stream;
+    std::vector<uint32_t> m_mjpeg_streams;
+    frame_list m_mjpeg_frames;
+    uint32_t m_movi_pos;
+    uint8_t  m_number_of_streams;
+    bool     m_is_indx_present;
+
+    bool     m_is_first_frame;
+    frame_iterator m_frame_iterator;
+    Mat m_current_frame;
+};
+
+bool MotionJpegCapture::parseMovi(MjpegInputStream&)
+{
+    //not implemented
+    return true;
+}
+
+uint32_t MotionJpegCapture::getFramePos() const
+{
+    if(m_is_first_frame)
+        return 0;
+
+    if(m_frame_iterator == m_mjpeg_frames.end())
+        return (uint32_t)m_mjpeg_frames.size();
+
+    return (uint32_t)(m_frame_iterator - m_mjpeg_frames.begin() + 1);
+}
+
+/*
+     -   **CV_CAP_PROP_FRAME_WIDTH** Width of the frames in the video stream.
+     -   **CV_CAP_PROP_FRAME_HEIGHT** Height of the frames in the video stream.
+     -   **CV_CAP_PROP_FPS** Frame rate.
+     -   **CV_CAP_PROP_FOURCC** 4-character code of codec.
+     -   **CV_CAP_PROP_FRAME_COUNT** Number of frames in the video file.
+     -   **CV_CAP_PROP_FORMAT** Format of the Mat objects returned by retrieve() .
+     -   **CV_CAP_PROP_MODE** Backend-specific value indicating the current capture mode.
+*/
+
+bool MotionJpegCapture::setProperty(int property, double value)
+{
+    if(property == CAP_PROP_POS_FRAMES)
+    {
+        if(int(value) == 0)
+        {
+            m_is_first_frame = true;
+            m_frame_iterator = m_mjpeg_frames.end();
+            return true;
+        }
+        else if(m_mjpeg_frames.size() > value)
+        {
+            m_frame_iterator = m_mjpeg_frames.begin() + int(value - 1);
+            m_is_first_frame = false;
+            return true;
+        }
+    }
+
+    return false;
+}
+
+double MotionJpegCapture::getProperty(int property) const
+{
+    switch(property)
+    {
+        case CAP_PROP_POS_FRAMES:
+            return (double)getFramePos();
+        case CAP_PROP_POS_AVI_RATIO:
+            return double(getFramePos())/m_mjpeg_frames.size();
+        case CAP_PROP_FRAME_WIDTH:
+            return 0;
```

Comment:
can you please implement FRAME_WIDTH, FRAME_HEIGHT and FPS properties? They are used quite often by users?